<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <arg name="publish_odom_tf" default="true"/>
    <arg name="robot_address" default="10.0.88.2"/>
    <!-- <arg name="robot_address" default="127.0.0.1"/> -->
    <arg name="robot_port" default="5800"/>

    <group ns="tj2">
        <node name="tj2_tunnel" pkg="tj2_tunnel" type="tj2_tunnel_node" output="screen" required="true">
            <rosparam param="joint_names" file="$(find tj2_description)/config/2022_robot_joints.yaml" command="load"/>
            <param name="host" value="$(arg robot_address)"/>
            <param name="remote_angular_units_conversion" value="1.0"/>
            <param name="remote_linear_units_conversion" value="1.0"/>
            <param name="min_linear_cmd" value="0.2"/>
            <param name="min_angular_z_cmd" value="1.0"/>
            <param name="zero_epsilon" value="0.01"/>
            <param name="port" value="$(arg robot_port)"/>
            <param name="cmd_vel_timeout" value="1.0"/>
            <param name="publish_odom_tf" value="true"/>
        </node>
    </group>

    <include file="$(find tj2_tunnel)/launch/twist_mux.launch"/>
</launch>
